### PR TITLE
refactor(console): improve error handling in admin console

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

Improved error handling in admin console:
* Session expired page re-uses AppError component, and now detailed error message is displayed as well.
* Improved full screen error page display logic, now the error screen will be displayed and the app won't hang at loading stage when error occurs.
* Will show error message from oidc provider on callback
* Whenever an error occurs, the swr will not repeatedly retry fetching the setting configs on app init.
* Bump @logto/react SDK to v0.1.11

image

image

image

## Testing

- [x] Session expired page now shows details message. See screenshot 1 and 2.
- [x] Callback page now receives and displays error messages from oidc provider
- [x] Swr will not repeatly sending requests to get settings when user auth failed.

### CODE REVIEW

File: packages/console/src/components/SessionExpired/index.tsx [hunk 1]
```diff
@@ -3,30 +3,31 @@ import React from 'react';
...
   const href = useHref('/callback');
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
 
   return (
-    <div className={styles.container}>
-      <img src={WarningIcon} />
-      <div className={styles.title}>{t('session_expire.title')}</div>
-      <div className={styles.subtitle}>{t('session_expire.subtitle')}</div>
+    <AppError
+      title={t('session_expire.title')}
```
Comment (gao-sun) [inline packages/console/src/components/SessionExpired/index.tsx:17]: expired

File: packages/console/src/components/AppContent/index.tsx [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { themeStorageKey } from '@/consts';
 import useAdminConsoleConfigs from '@/hooks/use-configs';
 import initI18n from '@/i18n/init';
 
+import AppError from '../AppError';
```
Comment (gao-sun) [inline packages/console/src/components/AppContent/index.tsx:10]: nit: better to use `@/components/`

File: packages/console/src/pages/Callback/index.tsx [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
-  useHandleSignInCallback(getBasename());
+  const { error } = useHandleSignInCallback(getBasename());
+
+  if (error) {
+    const errorMessage =
+      error instanceof LogtoError && error.data instanceof OidcError
+        ? error.data.errorDescription
+        : error.message;
+
+    return <AppError errorMessage={errorMessage} callStack={error.stack} />;
```
Comment (gao-sun) [inline packages/console/src/pages/Callback/index.tsx:17]: nit: show error code as well?

**Review by gao-sun — APPROVED** [review]
kool

File: packages/console/src/pages/Callback/index.tsx (hunk 3 above)
Comment (demonzoo) [reply packages/console/src/pages/Callback/index.tsx:17]: How about we only use `error` code when `errorMessage` is empty?

File: packages/console/src/components/SessionExpired/index.tsx (hunk 1 above)
Comment (demonzoo) [reply packages/console/src/components/SessionExpired/index.tsx]: updated, thanks

File: packages/console/src/components/AppContent/index.tsx (hunk 2 above)
Comment (demonzoo) [reply packages/console/src/components/AppContent/index.tsx]: Updated

File: packages/console/src/pages/Callback/index.tsx (hunk 3 above)
Comment (demonzoo) [reply packages/console/src/pages/Callback/index.tsx]: After short discussion, we'll add error code and the format would be `${errorCode}: ${errorMessage}`

**Review by gao-sun — APPROVED** [review]

**Pull request merged**